File: frontend/src/app/models/book-request.ts
import { TripLeg } from './trip-leg';

export class BookRequest {
  constructor(private accountNo: number,
    private totalFare: number,
    private bookingFee: number,
    private flightClass: string,
    private meal: string,
    private seatNo: string,
    private legs: TripLeg[],
    private fromLegNo: number) { }
}

File: backend/src/main/java/flightbook/entity/reservation/BookRequest.java
package flightbook.entity.reservation;

import flightbook.entity.leg.TripLeg;
import lombok.AllArgsConstructor;
import lombok.Data;

import java.util.List;

@Data
@AllArgsConstructor
public class BookRequest {
    private int accountNo;
	private double totalFare;
	private double bookingFee;
    private String flightClass;
    private String meal;
    private String seatNo;
	private List<TripLeg> legs;
	private int fromLegNo;
}
